lib/_http_outgoing.js

```diff
@@ -426,6 +430,26 @@ OutgoingMessage.prototype.getHeader = function getHeader(name) {
 };
 
 
+// Returns somewhat shallow copy of current outgoing headers. Array header
+// values will be cloned but no deeper cloning will occur.
+OutgoingMessage.prototype.getHeaders = function getHeaders(namesOnly) {
+  const headers = this._headers;
+  const keys = Object.keys(headers);
+  if (namesOnly)
+    return keys;
+  const ret = new OutgoingHeaders();
+  for (var i = 0; i < keys.length; ++i) {
+    const key = keys[i];
+    const val = headers[key][1];
```

Comment:
For my own curiosity, have you checked if `const` within a loop has any perf penalty vs a `var` outside the loop?